### PR TITLE
Remove session mutex, status variables atomic inc

### PR DESCRIPTION
**Pull request by max-hoffman**
Avoid concurrent ctx use in doQuery. I standardized the session status variables to uint64 to make it easier to have them be atomics. The return types for these variables in MySQL seem to be strings, so switching them from int640->uint64 seems OK. We could always change the presentation layer later. Removing the session mutex appears to be safe after these changes.

### CODE REVIEW

File: sql/base_session.go [hunk 1]
```diff
@@ -271,30 +236,19 @@ func (s *BaseSession) GetAllStatusVariables(_ *Context) map[string]StatusVarValu
...
-	statVal, ok := statVar.Val.(uint64)
-	if !ok {
-		return fmt.Errorf("status variable %s is not a uint64", statVarName)
-	}
 	if val < 0 {
-		statVar.Val = statVal - uint64(-val)
+		s.statusVars[statVarName].Increment(-(uint64(val)))
 	} else {
-		statVar.Val = statVal + uint64(val)
+		s.statusVars[statVarName].Increment(-(uint64(-val)))
```
Comment (jycor) [inline sql/base_session.go:245]: I think this and the above are flipped?
```go
if val < 0 {
    s.statusVars[statVarName].Increment(-(uint64(-val)))
} else {
    s.statusVars[statVarName].Increment((uint64(val)))
}
```

**Review by jycor — APPROVED** [review]
LGTM

File: sql/base_session.go (hunk 1 above)
Comment (max-hoffman) [reply sql/base_session.go:245]: yep good catch

File: sql/variables/status_variables.go [hunk 2]
```diff
@@ -138,67 +153,67 @@ var statusVars = map[string]sql.StatusVariable{
 		Name:    "Aborted_clients",
 		Scope:   sql.StatusVariableScope_Global,
 		Type:    types.NewSystemIntType("Aborted_clients", 0, 0, false),
-		Default: int64(0),
+		Default: atomic.Uint64{},
```
Comment (zachmu) [inline sql/variables/status_variables.go:156]: Better to make default values explicit  (`atomic.Uint64{0}`)

**Review by zachmu — APPROVED** [review]
LGTM

File: sql/variables/status_variables.go (hunk 2 above)
Comment (max-hoffman) [reply sql/variables/status_variables.go:156]: we would need to make a default object and load to make the zero explicit, i think
```go
defaultInc := atomic.Uint64{}
defaultInc.Store(0)
...
Default: defaultInc,
```
which seems more difficult?

**Pull request merged**